File: EventSignup/Models/ViewModels/PersonHeatModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EventSignup.Web.Models.ViewModels
{
    public class PersonHeatModel
    {
        public int id { get; set; }
        public string rxEvent { get; set; }
        public bool? partner { get; set; }
        public string comments {get;set;}
        public string partnerRxEvent { get; set; }
        public string partnerName { get; set; }
        public string partnerSex { get; set; }

        public PersonModel person { get; set; }
        public HeatModel heat { get; set; }        
    }
}

File: EventSignup/ClientApp/app/models/person-heat.model.ts
import { Heat } from './heat.model';
import { Person } from './person.model';


export class PersonHeat {
    id: number;
    rxEvent: string;
    partner: boolean;
    partnerRxEvent: string;
    partnerComments: string;
    partnerName: string;
    heat: Heat;
    person: Person;
    partnerSex: string;

    constructor() {
        this.heat = new Heat();
        this.person = new Person();
    }
}